### PR TITLE
[#11327] Compare table elements that are expected to be numbers numerically

### LINKED ISSUE #11327: Compare table elements that are expected to be numbers numerically
- **Environment**: `master` branch at commit 8a97db6

**Steps to reproduce**
Instructor views the result of a session question, like a rank recipients question with more than 10 possible ranks.

**Expected behaviour**
Table elements expected to be numbers are compared numerically and sorted in the correct order.

**Actual behaviour**
Screenshot 2021-07-31 at 10 26 21 PM

**Additional info**
Referring to file[ `table-comparator.service.ts`](https://github.com/TEAMMATES/teammates/blob/master/src/web/services/table-comparator.service.ts). Elements `RANK_RECIPIENTS_SELF_RANK`, `RANK_RECIPIENTS_OVERALL_RANK`, `RANK_RECIPIENTS_TEAM_RANK` etc. faces the above issue. For elements like `RANK_OPTIONS_OVERALL_RANK` and `MCQ_RESPONSE_COUNT`, while they are correctly sorted via natural ordering, it might be more appropriate to sort them numerically as well as they are expected to be numbers.

### PR DESCRIPTION
**Pull request by arsatis**
Fixes #11327

**Outline of Solution**

Under the file (table-comparator.service.ts)[https://github.com/TEAMMATES/teammates/blob/master/src/web/services/table-comparator.service.ts], the following elements were repositioned within the "switch" statement:

* MCQ_RESPONSE_COUNT
* MSQ_RESPONSE_COUNT
* RANK_OPTIONS_OVERALL_RANK
* RANK_RECIPIENTS_SELF_RANK
* RANK_RECIPIENTS_OVERALL_RANK
* RANK_RECIPIENTS_OVERALL_RANK_EXCLUDING_SELF
* RANK_RECIPIENTS_TEAM_RANK
* RANK_RECIPIENTS_TEAM_RANK_EXCLUDING_SELF

so that they will be compared numerically instead of lexicographically/naturally.

### CODE REVIEW

**Comment by jianhandev** [comment]
Hi @arsatis is this PR ready for review?

**Comment by arsatis** [comment]
> Hi @arsatis is this PR ready for review?

Yep 👍

**Review by jianhandev — APPROVED** [review]
LGTM, would be great if you can include some screenshots in the description of future PRs for easier reference :D

File: src/web/services/table-comparator.service.ts [hunk 1]
```diff
@@ -67,14 +67,11 @@ export class TableComparatorService {
       case SortBy.RUBRIC_CHOICE:
       case SortBy.RANK_RECIPIENTS_TEAM:
       case SortBy.RANK_RECIPIENTS_RECIPIENT:
-      case SortBy.RANK_OPTIONS_OVERALL_RANK:
       case SortBy.NUMERICAL_SCALE_MAX:
       case SortBy.NUMERICAL_SCALE_MIN:
```
[the remark this answers is not in the document]
Comment (arsatis) [reply src/web/services/table-comparator.service.ts:71]: > LGTM, would be great if you can include some screenshots in the description of future PRs for easier reference :D

Alright, will do that in future 👍

File: src/web/services/table-comparator.service.ts [hunk 2]
```diff
@@ -121,6 +119,8 @@ export class TableComparatorService {
       case SortBy.CONSTSUM_RECIPIENTS_POINTS:
       case SortBy.RUBRIC_WEIGHT_AVERAGE:
       case SortBy.RUBRIC_TOTAL_CHOSEN_WEIGHT:
+      case SortBy.NUMERICAL_SCALE_MAX:
+      case SortBy.NUMERICAL_SCALE_MIN:
```
Comment (arsatis) [inline src/web/services/table-comparator.service.ts:123]: Changes made

**Review by moziliar — APPROVED** [review]
LGTM

**Pull request merged**